test/blackbox-tests/test-cases/fmt/run.t

```diff
@@ -81,7 +81,86 @@ it gets wrapped.
 
 In multi-line strings, newlines are escaped.
 
-  $ dune unstable-fmt < multi-line-strings
+  $ dune unstable-fmt <<EOF
+  > (echo "\> multi
+  >       "\> line
+  >       "\> string
+  > )
+  > 
+  > (echo "\
+  > multi
+  > line
+  > string
+  > ")
+  > EOF
   (echo "multi\nline\nstring\n")
   
   (echo "multi\nline\nstring\n")
+
+Comments are preserved.
+
+  $ dune unstable-fmt <<EOF
+  > ; comment
+  > (;first comment
+  > a b;comment for b
+  > ccc;multi
+  > ;line
+  > ;comment for ccc
+  > d
+  > e
+  > ; unattached comment
+  > f
+  > ; unattached
+  > ; multi-line
+  > ; comment
+  > g
+  > )
+  > EOF
+  ; comment
+  
+  (; first comment
+   a
+   b ; comment for b
+   ccc ; multi
+       ; line
+       ; comment for ccc
+   d
+   e
+   ; unattached comment
+   f
+   ; unattached
+   ; multi-line
+   ; comment
+   g)
+
+When a comment is at the end of a list, the ")" is on a own line.
+
+  $ dune unstable-fmt <<EOF
+  > (a ;comment
+  > )
+  > (a ;multiline
+  > ;comment
+  > )
+  > EOF
```

Comment:
Should this be displayed as:

```
(a ; comment
 )

(a ; multiline
   ; comment
 )
```

?

It seems a bit odd to change the attachment rules for the last element.

Reply:
indeed, that's an error. I'll add a test case for this one as well (last attached / last unattached).

Reply:
done! and this simplifies the rules, as a bonus.